pkg/cloudprovider/gce/gce.go

```diff
@@ -115,7 +123,20 @@ func newGCECloud() (*GCECloud, error) {
 	if err != nil {
 		return nil, err
 	}
-	client := oauth2.NewClient(oauth2.NoContext, google.ComputeTokenSource(""))
+	tokenSource := google.ComputeTokenSource("")
+	if config != nil {
+		var cfg Config
+		if err := gcfg.ReadInto(&cfg, config); err != nil {
+			return nil, err
+		}
+		if cfg.Global.ProjectID != "" {
```

Comment:
It might be better to only set project id if you find both expected vars in the config object.